lib/Dialect/ESI/Passes/ESIBuildManifest.cpp

```diff
@@ -165,15 +174,34 @@ std::string ESIBuildManifestPass::json() {
   j.attribute("api_version", esiApiVersion);
 
   j.attributeArray("symbols", [&]() {
-    for (auto symInfo : mod.getBody()->getOps<SymbolMetadataOp>()) {
-      if (!symbols.contains(symInfo.getSymbolRefAttr()))
+    // First, gather all of the manifest data for each symbol.
+    DenseMap<SymbolRefAttr, SmallVector<IsManifestData>> symbolInfoLookup;
+    for (auto symInfo : mod.getBody()->getOps<IsManifestData>()) {
+      FlatSymbolRefAttr sym = symInfo.getSymbolRefAttr();
+      if (!sym || !symbols.contains(sym))
         continue;
+      symbolInfoLookup[sym].push_back(symInfo);
+    }
+
+    // Now, emit a JSON object for each symbol.
+    for (const auto &symNameInfo : symbolInfoLookup) {
```

Comment:
Ideally, some sort of human-readable documentation would also have been modified with this change to the manifest builder (in other words, where is the manifest spec?)